### PR TITLE
Сore: Replace "new Function" by JSON.parse

### PR DESCRIPTION
**Pull request by GoodDayForSurf**
Replace "new Function" by JSON.parse for CSP restrictions script-src=unsafe-eval

### CODE REVIEW

File: js/core/config.js [hunk 1]
```diff
@@ -44,13 +44,25 @@ const config = {
...
+                throw errors.Error('E3018', ex, optionsString);
+            }
         }
-    }
+    },
+};
+
+const normalizeToJSONString = (optionsString) => {
+    return optionsString
+        .replace(/([{,])[\s\n]*'([^']+)'[\s\n]*:/g, '$1"$2":') // replace quotes for single quoted keys
```
Comment (timbset) [inline js/core/config.js]: Isn't it better to replace all single quotes first (like we do it on the line 64) and then do not care about keys wrapped by single quotes?

File: testing/tests/DevExpress.core/config.tests.js [hunk 2]
```diff
@@ -88,3 +88,31 @@ QUnit.test('deprecated fields', function(assert) {
...
+            { 'dxTemplate': { 'name': 'title', test: { arr: ['a1', 'a2'] } } }
+        ],
+        [
+            '"dxTemplate": { "name": "title" }',
+            { 'dxTemplate': { 'name': 'title' } }
+        ],
+        [
+            '{"dxTemplate": { "name": "title" }}',
+            { 'dxTemplate': { 'name': 'title' } }
+        ]
```
Comment (timbset) [inline testing/tests/DevExpress.core/config.tests.js:122]: We check the `\n` symbol in regular expression, but there is no test with multiline template. Maybe add it?

**Review by timbset — CHANGES_REQUESTED** [review]

File: js/core/config.js (hunk 1 above)
Comment (GoodDayForSurf) [reply js/core/config.js]: reasonable..

File: testing/tests/DevExpress.core/config.tests.js (hunk 2 above)
Comment (GoodDayForSurf) [reply testing/tests/DevExpress.core/config.tests.js:122]: fixed

**Review by timbset — DISMISSED** [review]

**Review by timbset — APPROVED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Pull request merged**